### ISSUE TITLE
Requires only component

### ISSUE THREAD

**Issue by mauron85** [srjturner/abdicate#1]
Hi, firstly thank you for this minimalist but powerful ioc lib.

Is it possible to have @Requires only components (not providing anything them-self)?
For example I have defined Hapi route as component. Basically they depend on server, but don't provide anything.

```
/**
 * @Requires 'server'
 */
module.exports = function helloRoutes(server) {
  // Add the route
  server.route({
    method: 'GET',
    path: '/hello',
    handler: function(request, reply) {
      reply('hello world');
    }
  });
};
```

But it seems if there are more like this in one dir, only one is called and registered as hapi route. If I add  @Provides annotation for each route (file) that it is working just fine.

**Comment by srjturner** [comment]
Thanks for this, it turns out the default naming of functions without a @Provides annotation was broken, so two functions in the same folder conflicted. I've fixed that and added a test for it, so it should work now.

**Comment by srjturner** [comment]
Released in npm as version 1.0.12

**Comment by mauron85** [comment]
Nice. thank you
